### PR TITLE
Fix nonascii object names

### PR DESCRIPTION
**Pull request by m-rossi**
As a follow up to https://github.com/astropy/sphinx-automodapi/pull/153 I recently had another issue with nonascii characters. This time its the object name itself, for example the class name.

This PR adds the most popular umlauts (äöüß) to the regular expressions.

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! LGTM but second pair of eyes won't hurt.

Needs a change log: https://github.com/astropy/sphinx-automodapi/blob/main/CHANGES.rst

**Comment by m-rossi** [comment]
> Needs a change log: https://github.com/astropy/sphinx-automodapi/blob/main/CHANGES.rst

Done.

File: sphinx_automodapi/tests/cases/non_ascii/output/api/sphinx_automodapi.tests.example_module.nonascii.NonAsciiÄöüß.rst [hunk 1]
```diff
@@ -1,17 +1,17 @@
-NonAscii
-========
+NonAsciiÄöüß
+============
```
Comment (bsipocz) [inline sphinx_automodapi/tests/cases/non_ascii/output/api/sphinx_automodapi.tests.example_module.nonascii.NonAsciiÄöüß.rst:2]: While it shouldn't be an issue with utf8, I would add more more odd-ball character that manages to mess up systems that can handle latin-1 but not latin-2
```suggestion
NonAsciiÄöüßő
=============
```

File: sphinx_automodapi/tests/cases/non_ascii/output/api/sphinx_automodapi.tests.example_module.nonascii.NonAsciiÄöüß.rst [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
 .. currentmodule:: sphinx_automodapi.tests.example_module.nonascii
 
-.. autoclass:: NonAscii
+.. autoclass:: NonAsciiÄöüß
```
Comment (bsipocz) [inline sphinx_automodapi/tests/cases/non_ascii/output/api/sphinx_automodapi.tests.example_module.nonascii.NonAsciiÄöüß.rst:6]:
```suggestion
.. autoclass:: NonAsciiÄöüßő
```

File: sphinx_automodapi/tests/example_module/nonascii.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
-__all__ = ['NonAscii']
+__all__ = ['NonAsciiÄöüß']
 
 
-class NonAscii(object):
+class NonAsciiÄöüß(object):
```
Comment (bsipocz) [inline sphinx_automodapi/tests/example_module/nonascii.py:4]:
```suggestion
class NonAsciiÄöüßő(object):
```

File: sphinx_automodapi/tests/example_module/nonascii.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
-__all__ = ['NonAscii']
+__all__ = ['NonAsciiÄöüß']
```
Comment (bsipocz) [inline sphinx_automodapi/tests/example_module/nonascii.py:1]:
```suggestion
__all__ = ['NonAsciiÄöüßő']
```

File: sphinx_automodapi/tests/cases/non_ascii/output/index.rst.automodsumm [hunk 5]
```diff
@@ -11,5 +11,5 @@
 .. autosummary::
     :toctree: api
 
-    NonAscii
+    NonAsciiÄöüß
```
Comment (bsipocz) [inline sphinx_automodapi/tests/cases/non_ascii/output/index.rst.automodsumm:14]:
```suggestion
    NonAsciiÄöüßő
```

File: sphinx_automodapi/tests/cases/non_ascii/output/api/sphinx_automodapi.tests.example_module.nonascii.NonAsciiÄöüß.rst [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
    :show-inheritance:
 
    .. rubric:: Methods Summary
 
    .. autosummary::
 
-      ~NonAscii.get_ß
-      ~NonAscii.get_äöü
+      ~NonAsciiÄöüß.get_ß
+      ~NonAsciiÄöüß.get_äöü
```
Comment (bsipocz) [inline sphinx_automodapi/tests/cases/non_ascii/output/api/sphinx_automodapi.tests.example_module.nonascii.NonAsciiÄöüß.rst:14]:
```suggestion
      ~NonAsciiÄöüßő.get_ß
      ~NonAsciiÄöüßő.get_äöü
```

File: sphinx_automodapi/utils.py [hunk 7]
```diff
@@ -125,10 +125,10 @@ def find_autosummary_in_lines_for_automodsumm(lines, module=None, filename=None)
     """
     autosummary_re = re.compile(r'^(\s*)\.\.\s+autosummary::\s*')
     automodule_re = re.compile(
-        r'^\s*\.\.\s+automodule::\s*([A-Za-z0-9_.]+)\s*$')
+        r'^\s*\.\.\s+automodule::\s*([A-Za-zäüöÄÜÖß0-9_.]+)\s*$')
```
Comment (bsipocz) [inline sphinx_automodapi/utils.py:128]:
```suggestion
        r'^\s*\.\.\s+automodule::\s*([A-Za-zäüöÄÜÖßő0-9_.]+)\s*$')
```

File: sphinx_automodapi/utils.py [hunk 8]
```diff
@@ -125,10 +125,10 @@ def find_autosummary_in_lines_for_automodsumm(lines, module=None, filename=None)
...
     module_re = re.compile(
-        r'^\s*\.\.\s+(current)?module::\s*([a-zA-Z0-9_.]+)\s*$')
-    autosummary_item_re = re.compile(r'^\s+(~?[_a-zA-Z][a-zA-Z0-9_.]*)\s*.*?')
+        r'^\s*\.\.\s+(current)?module::\s*([a-zA-ZäüöÄÜÖß0-9_.]+)\s*$')
+    autosummary_item_re = re.compile(r'^\s+(~?[_a-zA-ZäüöÄÜÖß][a-zA-ZäüöÄÜÖß0-9_.]*)\s*.*?')
```
Comment (bsipocz) [inline sphinx_automodapi/utils.py:131]:
```suggestion
        r'^\s*\.\.\s+(current)?module::\s*([a-zA-ZäüöÄÜÖßő0-9_.]+)\s*$')
    autosummary_item_re = re.compile(r'^\s+(~?[_a-zA-ZäüöÄÜÖßő][a-zA-ZäüöÄÜÖßő0-9_.]*)\s*.*?')
```

**Review by bsipocz — APPROVED** [review]
Look all good, thank you so much @m-rossi!

One bike-shedding suggestion, take it or leave it.

**Comment by pllim** [comment]
Thanks, all!

**Comment by m-rossi** [comment]
@pllim may we can have a new release including this fix?

**Comment by pllim** [comment]
Re: release -- @bsipocz , what do you think? Anything else we should fit it before release?

**Comment by bsipocz** [comment]
Not sure how far https://github.com/astropy/sphinx-automodapi/pull/182 is from being finished, but either case, I'm not against tagging a release with what we have now.

**Comment by pllim** [comment]
I pinged Nathaniel over there but I doubt it is ready to go. I can do a release tomorrow unless someone else beats me to it. (And please remind me if I forget!)

**Comment by bsipocz** [comment]
OK, I'll try to see if I can get to it today, after all I'm already in release mode for other libs.

**Comment by pllim** [comment]
@bsipocz , did you get to it? If not, I can probably do it too. Lemme know. Thanks!

**Comment by pllim** [comment]
@m-rossi , v0.17.0 is released. Thanks for your patience!

**Comment by bsipocz** [comment]
Thanks, and sorry that I eventually didn't get to it sooner.

**Pull request merged**